client/asset/interface.go

```diff
@@ -79,6 +79,7 @@ type Wallet interface {
 	// externally. This method will only return funding coins, e.g. unspent
 	// transaction outputs.
 	FundingCoins([]dex.Bytes) (Coins, error)
+	FeeRateWithFallback() uint64
```

Comment:
The concept of a "fallback fee" is implementation specific, so if we do add a fee getter to the interface, it should just be named `FeeRate`, imo. As per my other comment, the client really shouldn't be trying to calculate fees when the server screws up, so I don't see a need for a fee rate getter yet anyway.

If we did want to provide a current fee rate for the UI, I would actually prefer it comes from the server somehow. I realize that is slightly more complicated since it involves API changes, but a single source of truth is a more important concern here, I think. I was thinking it could maybe be added to the `match_proof`.

Reply:
I generally agree.  This was mainly an "investigation PR" since Joe was getting a 0 fee from the server somehow -- I believe the server wasn't updated.  I don't think we need to go this route of exposing a fee rate method to the client wallet or providing a fee rate server api endpoint.

Reply:
> This was mainly an "investigation PR" 

I thought that might be the case. Sorry for jumping the gun. 

> Although if the server does screw up, the client is faced with the choice of inaction penalty or doing something sensible with the swap.

If the client knows the server screwed up, it should quit negotiation and opt for the penalty. That **is** the sensible thing to do with the swap, I think.